src/azure-cli-core/azure/cli/core/profiles/_shared.py

```diff
@@ -243,7 +243,7 @@ def default_api_version(self):
         }),
         ResourceType.MGMT_MSI: '2022-01-31-preview',
         ResourceType.MGMT_APPSERVICE: '2022-03-01',
-        ResourceType.MGMT_IOTHUB: '2021-07-02',
+        ResourceType.MGMT_IOTHUB: '2022-04-30-preview',
```

Comment:
Since there is no functional upgrade in this PR, may I ask why we need to upgrade the api-version to the new preview version?

Reply:
This is the newer api and the service (and portal) will be using this as the main api. The newer api (and sdk) provide cosmos db endpoints and hub certificate migration functionality.
The changes will be in the extension that will rely on this newer api. See https://github.com/Azure/azure-iot-cli-extension/pull/556 and https://github.com/Azure/azure-iot-cli-extension/pull/547

Having one api will increase consistency and reduce confusion among customers since they will not see hub certificate properties or cosmos db endpoints with the older api.

Reply:
On top of what @vilit1 said, some background...

Historically for IoT _most_ control plane functionality resides in Azure CLI and all data plane resides in the IoT extension. We are aiming to pull in more control plane to the extension, starting with upcoming root cert migration & cosmos db endpoint capability. Currently the extension relies on the IoT Hub control plane/mgmt SDK provided by Az CLI, so we need to increment it to move forward.

Reply:
Got it, thanks~